NuRadioMC/EvtGen/generator.py

```diff
@@ -460,9 +371,134 @@ def J(E):
     return energies
 
 
+def generate_vertex_positions(volume, proposal, attributes):
+    """
+    helper function that generates the vertex position randomly distributed in simulation volume. 
+    The relevant quantities are also saved into the hdf5 attributes
+    
+    Parameters
+    volume: dictionarty
+        dict specifying the volume
+    proposal: bool
+        specifies if secondary interaction via proposal are calculated
+    attributes: dicitionary
+        dict storing hdf5 attributes
+    """
+    n_events = attributes['n_events']
+    if("fiducial_rmax" in volume):  # user specifies a cylinder
+        if('fiducial_rmin' in volume):
+            attributes['fiducial_rmin'] = volume['fiducial_rmin']
+        else:
+            attributes['fiducial_rmin'] = 0
+        attributes['fiducial_rmax'] = volume['fiducial_rmax']
+        attributes['fiducial_zmin'] = volume['fiducial_zmin']
+        attributes['fiducial_zmax'] = volume['fiducial_zmax']
+
+        rmin = attributes['fiducial_rmin']
+        rmax = attributes['fiducial_rmax']
+        zmin = attributes['fiducial_zmin']
+        zmax = attributes['fiducial_zmax']
+
+        # We increase the radius of the cylinder according to the tau track length
+        if(proposal):
+            tau_95_length = get_tau_95_length(attributes['Emax'])
+            if("full_rmin" in volume):
+                rmin = volume['full_rmin']
+            else:
+                rmin = attributes['fiducial_rmin'] / 3.
+            if('full_rmax' in volume):
+                rmax = volume['full_rmax']
+            else:
+                full_rmax = tau_95_length + attributes['fiducial_rmax']
+            if('full_zmax' in volume):
+                zmax = volume['full_zmax']
+            else:
+                zmax = attributes['fiducial_zmax'] / 3.
+            if('full_zmin' in volume):
+                zmin = volume['full_zmin']
+            else:
+                zmin = attributes['fiducial_zmin'] - tau_95_length  # we have a minus sign here because the zmin coordinate is negative
```

Comment:
This is actually not a great choice. The tau range is ~ 100 km at ~ 100 EeV, and you need to keep in mind that the range in rock is reduced. This would simulate an enormous cylinder at high energies, but we know upgoing events at these energies are completely negligible.

Reply:
@daniel-zid what do you suggest to improve it?

Reply:
Rock is ~ 3 times denser than ice, although radiation length is only 26.54 g/cm^2 vs 36.08 in ice. Maybe use the radiation length as correction factor? But of course that depends on the configuration file used with Proposal, so I don't have an easy solution unless we scan the config file. 

This increase in volume can be computationally expensive unless we implement a proper cut, and there's not much to be gained from it except at low energies. And even so...